### ISSUE TITLE
Failed with some images

### ISSUE THREAD

**Issue by ycjing** [jrosebr1/color_transfer#4]
Hi. Thanks for the code! Impressive work!
I find that for some images, the code seems to produce the wrong output. Here is the example:
**source:**
![rock](https://user-images.githubusercontent.com/15648732/27523148-e4d02ef8-5a5d-11e7-9492-3831aa529554.jpg)
**target:** 
![ocean_day](https://user-images.githubusercontent.com/15648732/27523240-8b0b51bc-5a5e-11e7-9b9e-b9de7be49edf.jpg)
**output:**
![oceanout](https://user-images.githubusercontent.com/15648732/27523244-8caae190-5a5e-11e7-935e-5a559b3f43d0.jpg)

I will appreciate it if you can offer me some guidance. Thanks ahead of time!

**Comment by jrosebr1** [comment]
It's important to note that not all color transfers will look aesthetically pleasing. Although, looking at your output image it makes me wonder if the channels of the image are being ordered incorrectly (since OpenCV represents images in BGR order). There is an [open question from the original paper](https://github.com/jrosebr1/color_transfer/pull/3/commits/5ade879f7bf6731179c49a6b2cdffe4b94e5021f) regarding the ordering of the standard deviations during the scaling phase, so I would also swap the ordering and see if the results a more aesthetically pleasing. You'll also find the question discussed in [this closed issue as well](https://github.com/jrosebr1/color_transfer/issues/1).

**Comment by ycjing** [comment]
@jrosebr1 Thank you very much for your quick response! I will have a try.

**Comment by YuhengZhi** [comment]
This is probably because the author made a little mistake at line 43-46 of `__init__.py`
```
# scale by the standard deviations
l = (lStdTar / lStdSrc) * l
a = (aStdTar / aStdSrc) * a
b = (bStdTar / bStdSrc) * b
```
In fact it should be like:
```
# scale by the standard deviations
l = (lStdSrc / lStdTar) * l
a = (aStdSrc / aStdTar) * a
b = (bStdSrc / bStdTar) * b
```
p.s. I myself did something similar too.

**Comment by jrosebr1** [comment]
If the implementation is wrong then I must have misunderstood the original paper. Equation 11 of the original paper shows that the standard deviation of the target is divided by the standard deviation of the source (which is what the code does):

https://www.cs.tau.ac.il/~turkel/imagepapers/ColorTransfer.pdf

This implementation follows the paper as best as I can understand the authors' original intention. As I mentioned in a previous comment I suggest choosing the ordering that provides the most aesthetically pleasing results.

**Comment by zqwerty** [comment]
I agree with @YuhengZhi 
@jrosebr1 the source gives content and the target gives color

**Comment by AdamSpannbauer** [comment]
Might be a dead thread, but I wanted to share results of flipping the standard deviation ratio in the case of the 2 images @ycjing presented when opening the issue.

Using the ratio where the target is the numerator (`(lStdTar / lStdSrc)`) produces the output shown in the original post of this issue.  Flipping the ratio to have the source in the numerator (`(lStdSrc / lStdTar)`) gives the results shown below (which are a truer color transfer in this case).

![image](https://user-images.githubusercontent.com/16326083/40124523-c057b3c8-58f6-11e8-8aee-e39e2d8b8fce.png)

**Comment by jrosebr1** [comment]
Closed #5
